src/components/checkboxFilter/index.tsx

```diff
@@ -1,51 +1,11 @@
-import React, { ReactNode } from 'react';
+import React from 'react';
 
-import { Box, chakra } from '@chakra-ui/react';
+import { Box } from '@chakra-ui/react';
 
-import HighlightedText from '../text/HighlightedText';
-import Checkbox from '../checkbox';
-
-type Item<ItemKey> = {
-  key: ItemKey;
-  label: string;
-  facet: number;
-  isChecked: boolean;
-  image?: ReactNode;
-  highlightIndices?: ReadonlyArray<[number, number]>;
-};
-
-type State<ItemKey extends string> = { [key in ItemKey]: boolean };
-
-type Props<ItemKey extends string> = {
-  /**
-   * Name of the filter - only used on the DOM.
-   */
-  name: string;
-  /**
-   * @template ItemKey
-   * @param {Item} item     renders one checkbox { key: ItemKey, label: string, facet: number, isChecked: boolean }
-   * @param item.key        unique name of the checkbox
-   * @param item.label      label shown on the UI
-   * @param item.facet      Numeric value shown next to the checkbox label. Indicates how many search results are going to be visible after the checkbox has been applied.
-   * @param item.isChecked  The checkbox filter is a controlled component and the updated filter value must be passed in order to see the correct state.
-   * @param item.image      image/icon shown on the UI
-   */
-  items: Item<ItemKey>[];
-  /**
-   * Callback function that is triggered after any checkbox has been clicked.
-   * @param updatedItem     contains the modified checkbox with the new value
-   * @param newState        contains the new state of the whole filter group
-   */
-  onApply: (updatedItem: Item<ItemKey>, newState: State<ItemKey>) => void;
-  numberOfColumnsOnDesktop?: number;
-};
-
-const addThousandSeparatorToNumber = (value: number) => {
-  return new Intl.NumberFormat('de-CH').format(value);
-};
+import { Props } from './type';
+import CheckboxWithOptions from './CheckboxWithOptions';
 
 function CheckboxFilter<ItemKey extends string>({
-  name,
   items,
```

Comment:
i know this is a pretty big change for this PR, we can also discuss together. But I was kind of hoping to only have 1 compoent called CheckboxFilter at the end that takes a parent and a list of children. Based on that to render either a group or a flat list as we have now.
Given that all grouped checkbox filters are collapsible, this should be alright todo.

Secondly, CheckboxGroup already exists and for the flat checkbox list we use the base checkbox component for it. Can we check if we somehow can reuse this and just add a flag if it should be expandable or not?